### DISCUSSION TITLE
EvaluateExpressionAsync hangs indefinitely on some sites.

### DISCUSSION THREAD

**Discussion by spiderbait90** [hardkoded/puppeteer-sharp#1665] [Q&A]
This method hangs indefinitely on some sites....any ideas?
![image](https://user-images.githubusercontent.com/26646771/112471076-1f8d2a80-8d74-11eb-81a0-1aee296805e3.png)

**Comment by kblok** [comment]
Do you have a public site to test that?

**Reply by spiderbait90** [reply]
it hangs on this site => https://africanreporter.co.za/
Sometimes it hangs on others and sometimes it does not.

**Reply by spiderbait90** [reply]
![image](https://user-images.githubusercontent.com/26646771/112472925-5e23e480-8d76-11eb-9f68-9bdbbcaafbb8.png)
Debug pause.

**Reply by kblok** [reply]
I run this for a minute with no issues

```
while (true)
{
    await page.GoToAsync("https://africanreporter.co.za/");
    Console.WriteLine(string.Join("\n", await page.EvaluateExpressionAsync<string[]>("Array.from(document.querySelectorAll('a')).map(e => e.href)")));
}
```

**Reply by spiderbait90** [reply]
Can you also run it on each sub url in the main URL ?

**Reply by spiderbait90** [reply]
it hangs here on var response = await remote.ConfigureAwait(false);:
![image](https://user-images.githubusercontent.com/26646771/112485640-d47b1380-8d83-11eb-8f4b-edec0c5524b9.png)
